react/react1/week3/my-todo-list/src/components/TodoList.js

```diff
@@ -0,0 +1,68 @@
+import { useState, useEffect } from 'react';
+import Button from './Button';
+import TodoRow from './TodoRow';
+import AddTodo from './AddTodo';
+import Calendar from './Calendar';
+import BlackBorder from './BorderComponent';
+
+const TodoList = () => {
+  const [date, setDate] = useState('');
+  const [newTodo, setNewTodo] = useState('');
+  const [todosList, setTodosList] = useState([]);
+
+  const API_URL =
+    'https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw';
+
+  const FetchApi = () => {
+    fetch(API_URL)
+      .then((response) => response.json())
+      .then((result) => setTodosList(result));
+  };
+
+  useEffect(() => {
+    FetchApi();
+  }, []);
+
+  let uniqId = Math.random().toString(16).slice(2);
+
+  const newTodoItem = {
+    id: uniqId,
+    deadline: date,
+    description: newTodo,
+  };
+
+  function addTodo() {
+    setTodosList([...todosList, newTodoItem]);
+  }
+
+  function deleteTodoList(id) {
+    const updetedList = todosList.filter((el) => el.id !== id);
+
+    setTodosList(updetedList);
+  }
+
+  return (
+    <>
+      <AddTodo newTodo={newTodo} setNewTodo={setNewTodo} />
+      <Calendar date={date} setDate={setDate} />
+      <Button onClick={addTodo} bigSize text='Add todo' />
+      <div>
+        {todosList.map((todo) => (
+          <BlackBorder>
+            <TodoRow
```

Comment:
There is actually no reason to do the blackborder component, i would just do a div here with the styling

Reply:
I agree with you, but it was part of the homework
![image](https://user-images.githubusercontent.com/73549460/196433365-32dc51df-682e-4160-aa39-eca14deb3ac0.png)

Reply:
Ohhhh, i did not see that - i was wondering why some many people were doing it. My bad then! Sorry